### PR TITLE
Set relative

### PR DESCRIPTION
**Pull request by samwaseda**
`scaled_positions` had to be abolished for the following reasons:

1. It is not that commonly used
2. It is not possible to keep `positions` and `scaled_positions` synchronised, since when we try to change an element, e.g. via `positions[0,0] = 0.5`, the setter is not called and therefore we have no control over `scaled_positions`.
3. We would like to keep the functionality of `positions` as a numpy array.

Therefore, please use `set_scaled_positions` and `get_scaled_positions` in the future.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@samwaseda Can you also upload the notebook with the initial bug?

**Comment by samwaseda** [comment]
You want to see it? Otherwise it’s already included in the unit tests

**Comment by jan-janssen** [comment]
The notebook would be easier.

**Comment by jan-janssen** [comment]
I just realised that ASE also does not have `scaled_positions` and uses `set_scaled_positions` and `get_scaled_positions` as well. So as long as we are compatible, that sounds fine to me.

**Comment by samwaseda** [comment]
lattice = Project('.').create_structure('Al', 'fcc', 4)

basis_absolute = lattice.copy()
basis_absolute.set_absolute()
basis_relative = lattice.copy()
basis_relative.set_relative()

# basis_relative.positions[-1, 0] = 0.5
# basis_absolute.positions[-1, 0] = 0.5
basis_relative.scaled_positions[-1, 0] = 0.5
basis_absolute.scaled_positions[-1, 0] = 0.5

File: pyiron/atomistics/structure/atoms.py [hunk 1]
```diff
@@ -2955,10 +2910,20 @@ def rotate_euler(self, center=(0, 0, 0), phi=0.0, theta=0.0, psi=0.0):
         rcoords = np.dot(a, np.transpose(rcoords))
         # Move back to the rotation point
         if self._is_scaled:
-            self.scaled_positions = np.transpose(rcoords) + center
+            self.get_scaled_positions(np.transpose(rcoords) + center)
```
Comment (jan-janssen) [inline pyiron/atomistics/structure/atoms.py]: Should not this be set_scaled_positions ?

Comment (samwaseda) [reply pyiron/atomistics/structure/atoms.py]: brilliant comment -> I'm gonna fix it

File: pyiron/atomistics/structure/atoms.py [hunk 2]
```diff
@@ -1443,10 +1400,8 @@ def scaled_pos_xyz(self):
         Returns:
 
         """
-        x = self.scaled_positions[:, 0]
-        y = self.scaled_positions[:, 1]
-        z = self.scaled_positions[:, 2]
-        return x, y, z
+        xyz = self.get_scaled_positions()
+        return xyz[0], xyz[1], xyz[2]
```
Comment (jan-janssen) [inline pyiron/atomistics/structure/atoms.py]: Should not it be `xyz[:, 0], xyz[:, 1], xyz[:, 2]`

File: pyiron/atomistics/structure/atoms.py (hunk 1 above)
Comment (samwaseda) [reply pyiron/atomistics/structure/atoms.py]: so this was not just a problem of this line. This function anyway doesn't seem to work. I'm looking into it more in detail

**Pull request merged**